src/attributes.h

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2009-2010 Satoshi Nakamoto
+// Copyright (c) 2009-2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_ATTRIBUTES_H
+#define BITCOIN_ATTRIBUTES_H
+
+#if defined(__has_cpp_attribute) && __has_cpp_attribute(nodiscard)
+#  define NODISCARD [[nodiscard]]
+#elif defined(_MSC_VER) && _MSC_VER >= 1700
+#  define NODISCARD _Check_return_
+#else
+#  define NODISCARD __attribute__((warn_unused_result))
```

Comment:
This isn't guaranteed to be valid AFAIK?

Reply:
Do you have any suggestion on a more proper way to do it?

FWIW this is how Chromium does it :-)

```
#undef WARN_UNUSED_RESULT
#if defined(COMPILER_GCC) || defined(__clang__)
#define WARN_UNUSED_RESULT __attribute__((warn_unused_result))
#else
#define WARN_UNUSED_RESULT
#endif
```